### PR TITLE
Define simple vars on your operations

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

**Review by Prometee — APPROVED** [review]

File: docs/configure_your_operations.md [hunk 1]
```diff
@@ -474,6 +475,32 @@ class Book implements ResourceInterface
...
 | app_book_bulk_delete | DELETE          | /admin/books/bulk_delete |
 
+### Configure the vars
+
+It defines the simple vars that you can use on your templates.
+
+```php
+namespace App\Entity;
+
+use Sylius\Component\Resource\Metadata\Resource;
```
Comment (diimpp) [inline docs/configure_your_operations.md:486]: Missing
```php
use Sylius\Component\Resource\Metadata\Create;
```
for `#Create` attribute.

**Comment by diimpp** [comment]
Is this PR part of bigger initiative? For example, will be vars made translatable?

**Comment by Prometee** [comment]
> Is this PR part of bigger initiative? For example, will be vars made translatable?

The template is responsible of the translation, there is no need to manage translations in this bundle.

File: docs/configure_your_operations.md (hunk 1 above)
Comment (loic425) [reply docs/configure_your_operations.md:486]: done

**Review by oallain — APPROVED** [review]
Good work 👍

File: src/Component/Metadata/Resource/Factory/VarsResourceMetadataCollectionFactory.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        return $resourceCollectionMetadata;
+    }
+
+    private function addDefaults(ResourceMetadata $resource, HttpOperation $operation): Operation
+    {
+        if (null === $resourceVars = $resource->getVars()) {
+            return $operation;
+        }
+
+        return $operation->withVars(\array_merge($resourceVars, $operation->getVars() ?? []));
```
Comment (lchrusciel) [inline src/Component/Metadata/Resource/Factory/VarsResourceMetadataCollectionFactory.php:63]:
```suggestion
        return $operation->withVars(
            \array_merge($resource->getVars() ?? [], $operation->getVars() ?? [])
        );
```
or even
```suggestion
        return $operation->withVars($operation->getVars() ?? [] + $resource->getVars());
```

File: tests/Bundle/Command/DebugResourceCommandTest.php [hunk 3]
```diff
@@ -315,6 +317,7 @@ public function it_displays_the_metadata_for_given_resource_operation(): void
               routePrefix              null                      
               redirectToRoute          null                      
               redirectArguments        null                      
+              vars                     null
```
Comment (Zales0123) [inline tests/Bundle/Command/DebugResourceCommandTest.php:320]: Would be great to have those vars set up in one of these tests and check if they're properly displayed

**Review by lchrusciel — APPROVED** [review]

**Comment by Zales0123** [comment]
Thank you, Loïc! :1st_place_medal:

**Pull request merged**